### PR TITLE
report(viewer): minify inlined report-generator bundle

### PR DESCRIPTION
**Pull request by wardpeet**
**Summary**
Brfs inlines readFileSync functions from [report/html/html-report-assets.js](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/report/html/html-report-assets.js) inside viewer.js to generate an html report when clicking on saveHtml. 

I've added a [PR 94](https://github.com/browserify/brfs/pull/94) to enable options to transform the file content. I have my doubts that it will get accepted but i've already pushed it to my own npm account to get feedback on this change. I'm happy to publish this under a different namespace or packagename.

The new options allow us to run terser on our js files and compress them. We'll save **18.96KB**. The smaller the better is what I was thinking especially for a tool like Lighthouse 💪.

Before:
image

After:
image

This commit https://github.com/GoogleChrome/lighthouse/pull/9596/commits/e4bc45a04e29cf07c322f9f0ee4e75cd2735ca14 is the one you should look at.
This PR builds further on #9594

### CODE REVIEW

File: build/build-viewer.js [hunk 1]
```diff
@@ -106,7 +107,29 @@ async function compileJs() {
   // JS bundle from browserified ReportGenerator.
   const generatorFilename = `${sourceDir}/../lighthouse-core/report/report-generator.js`;
   const generatorBrowserify = browserify(generatorFilename, {standalone: 'ReportGenerator'})
-    .transform('brfs');
+    .transform('@wardpeet/brfs', {
+      /**
+       * @param {string} file
+       */
+      readFileSyncTransform: file => {
```
Comment (brendankenny) [inline build/build-viewer.js]: probably pull this out into a top-level named function

**Comment by connorjclark** [comment]
i love how you made webpack in browserify :)

**Comment by wardpeet** [comment]
@connorjclark it's now called wardpack :joy:
@brendankenny Unsure if it ever will be handled, Feel free to push it under another handle. I don't mind pushing it under a non scoped package name myself.

File: build/build-viewer.js [hunk 2]
```diff
@@ -98,6 +99,30 @@ async function html() {
   await safeWriteFileAsync(`${distDir}/index.html`, htmlSrc);
 }
 
+/**
+ * Minifies file which are read by fs.readFileSync (brfs)
+ *
+ * @param {string} file
+ */
+function minifyReadFileContent(file) {
```
Comment (wardpeet) [inline build/build-viewer.js]: I couldn't really find a proper name for this function, suggestions are welcome

**Comment by connorjclark** [comment]
Let's merrrrrrge this. I updated with master, and also minified the assets for the LR report generator. I verified the report still works with a terser pass (if we rely on function or class names, we'd have to pass special flags, but it seems we don't for the report).

Can confirm this saves a lot of bytes.

image

`dist/lightrider/report-generator-bundle.js` 244K -> 152K (38%)
`dist/viewer/src/viewer.js` 304K -> 208K (32%)

**Comment by wardpeet** [comment]
My legacy will live on! 🎉

**Pull request merged**